src/main/java/mate/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,44 @@
+package mate.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private static Set<String> allowedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        allowedUrls = new HashSet<>();
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse servletResponse, FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest req = (HttpServletRequest) servletRequest;
+        HttpSession session = req.getSession();
+        HttpServletResponse resp = (HttpServletResponse) servletResponse;
+        Long driverId = (Long) session.getAttribute("id");
+        if (driverId == null && allowedUrls.contains(req.getServletPath())) {
+            filterChain.doFilter(req, resp);
+            return;
+        }
+        if (driverId == null) {
+            resp.sendRedirect("/login");
+            return;
+        }
+        filterChain.doFilter(req, resp);
```

Comment:
can we simplify it?
<img width="736" alt="Screenshot 2021-12-14 at 09 45 55" src="https://user-images.githubusercontent.com/33026471/145994519-9e07d805-6edd-44be-b989-526f547b0f6a.png">

Reply:
yes